packages/world/src/WorldProxyFactory.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { Create2 } from "./Create2.sol";
+import { World } from "./World.sol";
+import { IWorldFactory } from "./IWorldFactory.sol";
+import { IBaseWorld } from "./codegen/interfaces/IBaseWorld.sol";
+import { IModule } from "./IModule.sol";
+import { ROOT_NAMESPACE_ID } from "./constants.sol";
+import { WorldProxy } from "./WorldProxy.sol";
+
+/**
+ * @title WorldProxyFactory
+ * @author MUD (https://mud.dev) by Lattice (https://lattice.xyz)
+ * @notice A factory contract to deploy new WorldProxy instances.
+ * @dev This contract allows users to deploy a new World and WorldProxy, install the InitModule, and transfer the ownership.
+ */
+contract WorldProxyFactory is IWorldFactory {
+  /// @notice Address of the init module to be set in the World instances.
+  IModule public immutable initModule;
+
+  /// @param _initModule The address of the init module.
+  constructor(IModule _initModule) {
+    initModule = _initModule;
+  }
+
+  /**
+   * @notice Deploys a new World and WorldProxy instance, installs the InitModule and transfers ownership to the caller.
+   * @dev Uses the Create2 for deterministic deployment.
+   * @param salt User defined salt for deterministic world addresses across chains
+   * @return worldAddress The address of the newly deployed WorldProxy contract.
+   */
+  function deployWorld(bytes memory salt) public returns (address worldAddress) {
+    // Deploy a new World and increase the WorldCount
+    uint256 _salt = uint256(keccak256(abi.encode(msg.sender, salt)));
+
+    // Deploy the world implementation
+    bytes memory worldBytecode = type(World).creationCode;
+    address worldImplementationAddress = Create2.deploy(worldBytecode, _salt);
```

Comment:
Could this be deployed once per world factory? And if it already exists we use the existing implementation?

Thinking about this, we could technically also undo the change of making `creator` not immutable and just require the deployer of the world proxy also to be the deployer of the implementation (which is the case when deploying via the world factory, and could be implemented this way by whoever wants to make a different world factory)

Reply:
Arguably we shouldn't even deploy the implementation in the factory, it's more of a parameter like `initModule`. I think for convenience it's fine but we could restructure it is a constructor argument and there is a `ensureWorldImplementation` in the deploy pipeline. 

I'm not sure the `creator` idea would work. The immutable variable is a constant on the implementation. If we `delegatecall` the implementation from the proxy, how would we read it?

Reply:
Because it's a constant you would have expose a read method or even read the bytecode 😬